File: Examples/Node.js/Files/DownloadFile.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var configProps = require('../config/config.json');
var data_path = '../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
var name = 'Rendering.docx';
try {	
	storageApi.PutCreate(name, versionId=null, storage=null, file= data_path + name , function(responseMessage) {
		assert.equal(responseMessage.status, 'OK', '');
		storageApi.GetDownload(name, versionId=null, storage=null, function(responseMessage) {
			console.log('status:', responseMessage.status);
		});
	});    

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/DotNET/CSharp/Files/DownloadFile.cs
using System;
using Com.Aspose.Storage.Api;
using Com.Aspose.Storage.Model;

namespace Files
{
    class DownloadFile
    {
        public static void Run()
        {
            // ExStart:1           
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            string Path = "Book1.xlsx";
            string versionId = null;
            string storage = null;            
            byte[] file = System.IO.File.ReadAllBytes(Common.GetDataDir() + Path);
            try
            {
                // Invoke Aspose.Storage Cloud SDK API to upload and download file
                storageApi.PutCreate(Path, versionId, storage, file);
                Com.Aspose.Storage.Model.ResponseMessage storageRes = storageApi.GetDownload(Path, versionId, storage);                
                System.Diagnostics.Debug.WriteLine("response:" + storageRes.ResponseStream);
                System.IO.File.WriteAllBytes(Common.GetDataDir() + Path, storageRes.ResponseStream);
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}
